pybop/observers/unscented_kalman.py

```diff
@@ -27,20 +26,65 @@ class UnscentedKalmanFilterObserver(Observer):
         To remove a state from the filter, set the corresponding row and col to zero in both sigma0 and process.
     measure : np.ndarray | float
         The covariance matrix of the measurement noise. If a float is provided, the covariance matrix is set to measure * np.eye(m), where m is the number of measurements.
+    dataset : Dataset
+        Dataset object containing the data to fit the model to.
+    check_model : bool, optional
+        Flag to indicate if the model should be checked (default: True).
+    signal: str
+        The signal to observe.
+    init_soc : float, optional
+        Initial state of charge (default: None).
+    x0 : np.ndarray, optional
+        Initial parameter values (default: None).
     """
 
     Covariance = np.ndarray
 
     def __init__(
         self,
+        parameters: List[Parameter],
         model: BaseModel,
-        inputs: Inputs,
-        signal: List[str],
         sigma0: Union[Covariance, float],
         process: Union[Covariance, float],
         measure: Union[Covariance, float],
+        dataset=None,
+        check_model=True,
+        signal=["Voltage [V]"],
+        init_soc=None,
+        x0=None,
     ) -> None:
-        super().__init__(model, inputs, signal)
+        super().__init__(parameters, model, check_model, signal, init_soc, x0)
+        if dataset is not None:
+            self._dataset = dataset.data
+
+            # Check that the dataset contains time and current
+            for name in ["Time [s]", "Current function [A]"] + self.signal:
```

Comment:
why does the dataset need "Current function [A]", the model might not include it or need it (e.g. exponential decay). Also this seems like something that should be defined on the dataset class (e.g. `self._dataset.check_valid()`)

Reply:
Great minds... I've moved the dataset checks to the Dataset class on another branch (PR #178) so here I just copied what was previously in the Problem class. The current function requirement still needs to be relaxed though.